packages/salesforce-adapter/src/constants.ts

```
@@ -48,6 +48,8 @@ export enum FIELD_TYPE_NAMES {
   LOOKUP = 'Lookup',
   MASTER_DETAIL = 'MasterDetail',
   ROLLUP_SUMMARY = 'Summary',
+  // internal-only placeholder for fields whose type is unknown
+  UNKNOWN = 'Unknown',
```

@ori-moisis based on our conversation I didn't add defenses against deploy attempts and will let them fail on salesforce errors (`soapenv:Client: 'Unknown' is not a valid value for the enum 'FieldType'`) - lmk if that changes.

---

I think we might need to filter out these fields when the user tries to create a new CustomObject for example because while I generally think this is a good enough error if the user explicitly tries to change the field, I don't think failing like this makes sense if the user tried to create a new custom object (most likely this can happen if the user tries to move a custom object from one environment to another).
The [change validator mechanism](https://github.com/salto-io/salto/blob/master/packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts) should be a good fit for this kind of filtering (and it will also give us the opportunity to give a more readable error message)